identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-enrol.jsp

```diff
@@ -0,0 +1,417 @@
+
+        function talkToDevice(){
+            var regRequest = '<%=Encode.forJavaScriptBlock(regRequest)%>';
+            var jsonRegRequest = JSON.parse(regRequest);
+            console.log(jsonRegRequest);
+            console.log(jsonRegRequest.publicKeyCredentialCreationOptions);
+
+            navigator.credentials.create({
+                publicKey: decodePublicKeyCredentialCreationOptions(jsonRegRequest.publicKeyCredentialCreationOptions),
+            })
+            .then(function(data) {
+                payload = {};
+                payload.requestId = jsonRegRequest.requestId;
+                payload.credential = responseToObject(data);
+                var reg = document.getElementById('challengeResponse');
+                reg.value = JSON.stringify(payload);
+
+                showKeynameFlow();
+            })
+            .catch(function(err) {
+                showError();
+                fidoError = err;
+            });
+        }
+
+        function retry() {
+            showFidoFlow();
+            talkToDevice();
+        }
+
+        function showError() {
+            $("#fido-header-error").show();
+            $("#fido-error-content").show();
+            $("#fido-header").hide();
+            $("#fido-initialize").hide();
+            $("#loader-bar").hide();
+            $("#fido-keyname-header").hide();
+            $("#fido-keyname-content").hide();
+        }
+
+        function showFidoFlow() {
+            $("#fido-header-error").hide();
+            $("#fido-error-content").hide();
+            $("#fido-header").show();
+            $("#fido-initialize").show();
+            $("#loader-bar").show();
+            $("#fido-keyname-header").hide();
+            $("#fido-keyname-content").hide();
+        }
+
+        function showKeynameFlow() {
+            $("#fido-header-error").hide();
+            $("#fido-error-content").hide();
+            $("#fido-header").hide();
+            $("#fido-initialize").hide();
+            $("#loader-bar").hide();
+            $("#fido-keyname-header").show();
+            $("#fido-keyname-content").show();
+        }
+
+        function cancelFlow(){
+            var form = document.getElementById('form');
+            var reg = document.getElementById('challengeResponse');
+            reg.value = JSON.stringify({ errorCode: 400, message: fidoError });
+            var scenario = document.getElementById('scenario');
+            scenario.value = "CANCEL_FIDO_ENROL" ;
+            form.submit();
+        }
+
+        function finishFidoFlow() {
+
+            var keynameInput = document.getElementById('keynameInput');
+            var displayName = document.getElementById('displayName');
+            var scenario = document.getElementById('scenario');
+
+            // Get the keyname input value and trim whitespace
+            var keyname = keynameInput.value.trim();
+
+            // Check if the keyname is not empty
+            if (keyname === '') {
```

Comment:
Comment and code doesn't match :)

Reply:
Resolved via https://github.com/wso2/identity-apps/pull/4156/commits/8968e9e9e80de6f997577d12f3f6222db508b1de